include/libxnvme_spec.h

```diff
@@ -2324,16 +2291,38 @@ struct xnvme_spec_nvm_idfy_ctrlr {
 XNVME_STATIC_ASSERT(sizeof(struct xnvme_spec_nvm_idfy_ctrlr) == 4096, "Incorrect size")
 
 /**
- * @todo Document this
+ * Representation of NVMe completion result for NVM command set I/O command set
+ * specific Identify Namespace
+ *
+ * That is, for opcode XNVME_SPEC_OPC_IDFY(0x06) with XNVME_SPEC_IDFY_NS_IOCS(0x05)
+ *
+ * @struct xnvme_spec_nvm_idfy_ns
  */
 struct xnvme_spec_nvm_idfy_ns {
-	uint8_t byte0_73[74];
+	uint64_t lbstm; ///< Logical block storage tag mask
+
+	/** Protection information capabilities */
+	union {
+		struct {
+			/** 16b guard protection information storage tag mask */
+			uint8_t gpistm    : 1;
+
+			/** 16b guard protection information storage tag support */
+			uint8_t gpists    : 1;
 
-	uint16_t mssrl; ///< Maximum Single Source Range Length
-	uint32_t mcl;   ///< Maximum Copy Length
-	uint8_t msrc;   ///< Maximum Source Range Count
+			/** storage tag check read support */
+			uint8_t stcrs     : 1;
```

Comment:
Order doesn't look right.
Bit 0 is for storage tag support, 1 is for storage tag mask and 2 is for storage tag check read support.
https://nvmexpress.org/wp-content/uploads/NVM-Express-NVM-Command-Set-Specification-1.0d-2023.12.28-Ratified.pdf

Reply:
Nice catch, thanks!